### PR TITLE
[MBL-821] Upgrade profile activity to RX2

### PR DESCRIPTION
**Pull request by mtgriego**
# 📲 What

Remove RX1 and replace it with RX2 in profile activity/viewmodel

# 🤔 Why

RX1 is old and busted, we need the new hotness

# 📋 QA

Confirm profile activity still works as normal

# Story 📖

[MBL-821](https://kickstarter.atlassian.net/browse/MBL-821)

[MBL-821]: https://kickstarter.atlassian.net/browse/MBL-821?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/activities/ProfileActivity.kt [hunk 1]
```diff
@@ -2,126 +2,138 @@ package com.kickstarter.ui.activities
...
 import com.kickstarter.ui.IntentKey
 import com.kickstarter.ui.adapters.ProfileAdapter
 import com.kickstarter.ui.extensions.loadCircleImage
+import com.kickstarter.ui.extensions.startActivityWithTransition
 import com.kickstarter.viewmodels.ProfileViewModel
+import io.reactivex.disposables.CompositeDisposable
 
-@RequiresActivityViewModel(ProfileViewModel.ViewModel::class)
-class ProfileActivity : BaseActivity<ProfileViewModel.ViewModel>() {
+class ProfileActivity : ComponentActivity() {
```
Comment (leighdouglas) [inline app/src/main/java/com/kickstarter/ui/activities/ProfileActivity.kt:27]: are we clearing the disposables anywhere here?

File: app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.kt [hunk 2]
```diff
@@ -80,13 +82,15 @@ interface ProfileViewModel {
         fun userNameTextViewText(): Observable<String>
     }
 
-    class ViewModel(environment: Environment) : ActivityViewModel<ProfileActivity>(environment), ProfileAdapter.Delegate, Inputs, Outputs {
-        private val client = requireNotNull(environment.apiClient())
-        private val currentUser = requireNotNull(environment.currentUser())
+    class ProfileViewModel(environment: Environment) : ViewModel(), ProfileAdapter.Delegate, Inputs, Outputs {
```
Comment (leighdouglas) [inline app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.kt:85]: same question as above, we should be clearing the disposables when the viewmodel is destroyed

**Review by leighdouglas — COMMENTED** [review]
Left a

File: app/src/main/java/com/kickstarter/ui/activities/ProfileActivity.kt (hunk 1 above)
Comment (mtgriego) [reply app/src/main/java/com/kickstarter/ui/activities/ProfileActivity.kt:27]: good catch, forgot about that

**Review by Arkariang — APPROVED** [review]

**Pull request merged**